docs/content/5.recipes/4.custom-session-data.md

```diff
@@ -1,57 +1,121 @@
+   ... // other config
+
+```
+We can see the new fields in the return object `u` `role` and `access_token`. Now we need to pass them in the callbacks.
+
+## Add callbacks to the `NuxtAuthProvider` to alter the session data
 
 First, you must add jwt and session callbacks to [the `NuxtAuthHandler`](/nuxt-auth/configuration/nuxt-auth-handler):
 ```ts
 // file: ~/server/api/auth/[...].ts
 export default NuxtAuthHandler({
+...
   callbacks: {
     // Callback when the JWT is created / updated, see https://next-auth.js.org/configuration/callbacks#jwt-callback
-    async jwt({ token, account, profile }) {
-      // Persist the OAuth access_token and or the user id to the token right after signin
-      if (account) {
-        token.accessToken = account.access_token
-        token.id = profile.id
+    jwt: async ({token, user}) => {
+      const isSignIn = user ? true : false;
+      if (isSignIn) {
+        token.jwt = user ? (user as any).access_token || '' : '';
+        token.id = user ? user.id || '' : '';
+        token.role = user ? (user as any).role || '' : '';
       }
-      return token
+      return Promise.resolve(token);
     },
-
     // Callback whenever session is checked, see https://next-auth.js.org/configuration/callbacks#session-callback
-    async session({ session, token, user }) {
-      // Send properties to the client, like an access_token and user id from a provider.
-      session.accessToken = token.accessToken
-      session.user.id = token.id
-
-      return session
-    }
+    session: async ({session, token}) => {
+      (session as any).role = token.role;
+      (session as any).uid = token.id;
+      return Promise.resolve(session);
+    },
   },
   ..., // other config
   providers: [...], // Your provider config
 })
 ```
+Notice that in the `jwt` callback we pass both the access_token and the role but in the session callback we leave out the access_token and we only pass the role since this will expose jwt on the frontend through the session object. This would be the recommendable approach, but you can pass it to the session if needed.
 
-## Updating TypeScript Interfaces for Accurate Typing
+After that you can access the new fields in the data portion of the session object:
 
-```ts
-// file: ~/next-auth.d.ts
-import { DefaultSession } from "next-auth"
+```
+const { status, data } = useSession(); 
+
+console.log(data);
+
+```
 
-declare module "next-auth" {
-  interface Session {
-    // New access token field we added to the session
-    accessToken: string
-    user: DefaultSession["user"]
-  }
-}
+output:
 ```
+{ "user": { "name": "xxxxxxx", "email": "xxxxx@xxxxx.gr" }, "expires": "2023-02-28T20:36:12.584Z", "role": "admin", "uid": 1 }
+```
+
+The access_token can be access as:
+
+```
+import { getToken } from '#auth'
+
+const token = await getToken({ event })
+
+console.log(token.jwt);
+```
+
+Please Notice that with this approach there is no need to use profile function or override the types next-auth.
```

Comment:
oh wow, that's nice, let's phrase it differently though as futurre readers will not know about the past approach:
```md
With this approach `data` will automatically be correctly typed.
```

Reply:
Thank you, I will make the changes tomorrow